### ISSUE TITLE
DC-Info-GAN initial implementation and results.

### ISSUE THREAD

**Issue by elggem** [opencog/semantic-vision#1]
I have commited an implementation of DC-Info-GAN in the repository [here](https://github.com/opencog/semantic-vision/tree/master/experimental-code/dc-info-gan). I've tried to write it in such a way to allow easy experimentation with this from now on. Please comment here if there are any questions and suggestions!

Right now I have only integrated the proper loss function for the discrete distributions, and this is working. For continuous variables this is not working great yet. These are the results for training 10 epochs:

![Animated Results](https://github.com/opencog/semantic-vision/raw/master/experimental-code/dc-info-gan/results/mnist-animation.gif)

Here is my next steps:

  * [x] Make Generation of images adapt to different latent specs
  * [x] Implement Mutual Information Loss for continous latent variables. I had a discussion with Ben on this and I think I remember that maybe MSE is actually enough here... More on this soon.
  * [ ] More experimentation with different latent specs / network configurations / loss functions, etc.
  * [ ] Work towards integrating with BEGAN.
  * [ ] Integrating TensorBoard for better evaluation.

**Comment by elggem** [comment]
I implemented more adaptive behaviour for continous probability distributions in the [development branch](https://github.com/opencog/semantic-vision/blob/development/experimental-code/dc-info-gan/gan.py). I am evaluating now wether or not we can just use MSE for this loss. This is according to 

http://aiden.nibali.org/blog/2016-12-01-implementing-infogan/

Will update here once I am happy about the results. The code already is much more adaptive for different latent codes, but it is not capturing the continous variables well...
